### PR TITLE
Introduce tagged-type definitions

### LINKED ISSUE #74: tagged-concise-mid-tag and other tagged type definitions needed
There should be tagged `tagged-` definitions for the various types of tags that can be included in a corim.
for example a corim can contain these CBOR tagged structures:
```
$concise-tag-type-choice /= #6.505(bytes .cbor concise-swid-tag)
$concise-tag-type-choice /= #6.506(bytes .cbor concise-mid-tag)
$concise-tag-type-choice /= #6.508(bytes .cbor concise-bom-tag)
```
but we don't have a way to refer to each as a CBOR tagged statement, such as:
```
tagged-concise-swid-tag = #6.505(bytes .cbor concise-swid-tag)
tagged-concise-mid-tag = #6.506(bytes .cbor concise-mid-tag)
tagged-concise-bom-tag = #6.508(bytes .cbor concise-bom-tag)
```
The $concise-tag-type-choice definitions could become:
```
$concise-tag-type-choice /= tagged-concise-swid-tag
$concise-tag-type-choice /= tagged-concise-mid-tag
$concise-tag-type-choice /= tagged-concise-bom-tag
```

### PR DESCRIPTION
**Pull request by yogeshbdeshpande**
Fixes #74

### CODE REVIEW

**Review by nedmsmith — APPROVED** [review]

File: cddl/cbor-tags.txt [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
-$concise-tag-type-choice /= #6.506(bytes .cbor concise-mid-tag)
-$concise-tag-type-choice /= #6.508(bytes .cbor concise-bom-tag)
+corim = tagged-concise-rim-type-choice
+$concise-rim-type-choice /= tagged-corim-map
+$concise-rim-type-choice /= tagged-signed-corim
+$concise-tag-type-choice /= tagged-concise-swid-tag
+$concise-tag-type-choice /= tagged-concise-mid-tag
+$concise-tag-type-choice /= tagged-concise-bom-tag
+$raw-value-type-choice /= tagged-bytes
+tagged-bytes = #6.560(bytes)
```
[the remark this answers is not in the document]
Comment (yogeshbdeshpande) [reply cddl/cbor-tags.txt]: Somehow github issue @nedmsmith 
I regenerated the file but it was not showing up properly.

I have now re-submitted and it all looks good!

[the remark this answers is not in the document]
Comment (thomas-fossati) [reply cddl/cbor-tags.txt]: > Your point is this is correct/incorrect?

incorrect, but we've fixed it now

**Comment by thomas-fossati** [comment]
we need to also adjust prose in the following sections/tables:
* §7.2 "New CBOR Tags"
* §2.1.2 "Tags"

File: draft-ietf-rats-corim.md [hunk 2]
```diff
@@ -1280,14 +1280,14 @@ IANA is requested to allocate the following tags in the "CBOR Tags" registry {{!
 
 |     Tag | Data Item           | Semantics                                                            | Reference |
 |     --- | ---------           | ---------                                                            | --------- |
-|     500 | `tag`               | A corim, one out of $concise-rim-type-choice, see {{sec-corim-tags}} | {{&SELF}} |
-|     501 | `map`               | A corim-map, see {{sec-corim-map}}                                   | {{&SELF}} |
-|     502 | `tag`               | A signed-corim, see {{sec-corim-signed}}                             | {{&SELF}} |
+|     500 | `tag`               | A tagged-concise-rim-type-choice, see {{sec-corim-tags}} | {{&SELF}} |
```
Comment (thomas-fossati) [inline draft-ietf-rats-corim.md:1283]: thanks for addressing my comment.  Please keep the existing typographic alignment though.

File: draft-ietf-rats-corim.md [hunk 3]
```diff
@@ -363,7 +363,7 @@ identifiers. Other types of identifiers could be defined as needed.
 ### Tags {#sec-corim-tags}
 
 A `$concise-tag-type-choice` is a tagged CBOR payload that carries either a
-CoMID ({{sec-comid}}) or a CoSWID {{-coswid}}.
+CoMID ({{sec-comid}}), or a CoSWID {{-coswid}}, or a CoBOM {{sec-cobom}}.
```
[the remark this answers is not in the document]
Comment (yogeshbdeshpande) [reply draft-ietf-rats-corim.md:366]: I checked online and the description:
CoMID ({{sec-comid}}), or a CoSWID {{-coswid}}, or a CoBOM {{sec-cobom}}.
is also correct and is written sometimes to be more explicit to state that the value can be either one of them

**Pull request merged**